src/lib.rs

```diff
@@ -637,6 +623,21 @@ impl Build {
         let out_dir = self.get_out_dir()?;
         let src = self.ensure_check_file()?;
         let obj = out_dir.join("flag_check");
+        let target = self.get_target()?;
```

Comment:
Let's move these calls into try_get_compiler and is_flag_supported, that would save a few calls in try_get_compiler

Reply:
Except that these are post cache check, so moving them earlier hurts the fast path (even if we can amortize them by moving them out of the loop in `try_get_compiler`). An alternative is to make them `&mut Option<_>` so that they can be remembered if they are set in the loop, but I'm not sure that's worth the complexity it introduces.

Reply:
That's true, they are probably cheap enough to not be an issue.